privilege/privileges/privileges.go

```diff
@@ -161,6 +161,33 @@ func (p *UserPrivileges) RequestVerificationWithUser(db, table, column string, p
 	return mysqlPriv.RequestVerification(roles, user.Username, user.Hostname, db, table, column, priv)
 }
 
+func (p *UserPrivileges) isValidHash(record *UserRecord) bool {
+	pwd := record.AuthenticationString
+	if len(pwd) == 0 {
```

Comment:
AFAIK there are some linters which prefer checking an empty string as `pwd == ""`. It's not currently enforced in our linter suite but may be in future.